src/Ladybug_SolarEnvelopeAdvanced.py

```diff
@@ -0,0 +1,160 @@
+# Solar Envelope
+# Provides an envelope (as a list of points) which represents the area in which new masses could be placed withought
+# interfering with the sun rights of surrounding buildings, given a filtered list of suns
+# By Boris Plotnikov with the assistance and guidance of Prof. Guedi Capeluto
+# pborisp@gmail.com
+# Ladybug started by Mostapha Sadeghipour Roudsari is licensed
+# under a Creative Commons Attribution-ShareAlike 3.0 Unported License.
+"""
+Use this component to generate a solar envelope for a given test surface, set of solar vectors, and context geometry that you want to ensure solar access to.  Solar envelopes are typically used to illustrate the volume that can be built within in order to ensure that a new development does not shade the surrounding properties for a given set of sun vectors.
+
+-
+Provided by Ladybug 0.0.59
+    
+    Args:
+        _baseSrf: A surface representing the area for which you want to create the solar envelope.
+        _obstacleCurves: A list of curves indicating the bottom borders of your surroundings for which you would like solar access to be kept.
```

Comment:
Thought that people could plug in the base surface for the obstacle curves, but this is simpler and more intuitive, done